src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/Fdc3AppExtensions.cs

```diff
@@ -0,0 +1,90 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+using Finos.Fdc3.AppDirectory;
+using Finos.Fdc3.Context;
+using MorganStanley.ComposeUI.Fdc3.DesktopAgent.Infrastructure.Internal;
+using MorganStanley.ComposeUI.Fdc3.DesktopAgent.Protocol;
+using AppMetadata = MorganStanley.ComposeUI.Fdc3.DesktopAgent.Protocol.AppMetadata;
+using Icon = MorganStanley.ComposeUI.Fdc3.DesktopAgent.Protocol.Icon;
+
+namespace MorganStanley.ComposeUI.Fdc3.DesktopAgent;
+
+public static class Fdc3AppExtensions
+{
+    /// <summary>
+    /// Determines if the app can accept the desired context in response to a raised intent.
+    /// </summary>    
+    internal static bool DoesAcceptContextType(this FlatAppIntent app, string contextType)
+    {
+        var contexts = app!.Intent?.Contexts;
+        if (contexts == null || !contexts.Any()) { return contextType == ContextType.Nothing.Type; }
```

Comment:
For me it's pretty weird to have the if statement and the act in one line contained by the brackets.

Reply:
In case of return statements I'm used to it, but I'll change it, I don't mind either way :)